### ISSUE TITLE
Add "previous" and "next" buttons for browsing the database

### ISSUE THREAD

**Issue by GoogleCodeExporter** [De7vID/klingon-assistant#13]
```
The feature request is to add buttons for "previous" and "next" which would go 
to the adjacent entries in the database, e.g., from {nach} "next" would go to 
{naD}.
```

Original issue reported on code.google.com by `De.vID.jonpIn` on 3 Jan 2013 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
+ maybe the possibility to "swipe" instead of buttons.
```

Original comment by `Lieven.l...@gmail.com` on 8 Jan 2013 at 2:15

**Comment by GoogleCodeExporter** [comment]
```
"Swipe" is impossible in the backport (so it's blocked on issue 46).
```

Original comment by `De.vID.jonpIn` on 10 Feb 2013 at 9:20

**Comment by LievenLitaer** [comment]
_ping_ I still would like a button to browse the dictionary by clicking "back" and "next".

**Comment by De7vID** [comment]
Where do you suggest the buttons should go in the UI?

**Comment by LievenLitaer** [comment]
At the bottom. When using a text, the length may vary too much (see "previous" vs. "next"), so I'd suggest arrow symbols. You could make them smaller, so you have some space for another button (perhaps the "speak" button?)
![boqwi _buttons](https://cloud.githubusercontent.com/assets/11874228/7083657/ca7097c6-df64-11e4-8c0d-aceda3020efc.png)

**Comment by De7vID** [comment]
What's the usefulness of doing this? I think it takes up too much screen space, and it doesn't do anything useful which can't already be done with existing functionality. If you want to go through the entries beginning with {gho}, for example, a search on {gho} already gives the list of results, which you can move between by using the list. What does this add?

The database is not organised sequentially, but for fast prefix searching. Linking each entry to a "previous" and "next" entry would involve either increasing the size of the database (literally putting pointers to previous and next entries within each entry) or significantly slow down its speed (each query would also have to find the previous and next entries).

**Comment by LievenLitaer** [comment]
<b>1. Usefulness:</b> I sometimes sit around (in the bus, on the puchpa') and think like "let's practice some klingon", by clicking around in the app. I randomly jump from page to page, but suddenly notice that I'm going circles, coming back to the same word.
<b>2. What does this add?</b> Browsing students may find entries they would otherwise have never met. It makes it easier to browse, because what you suggested requires to re-open the search window all the time, instead of just browsing. When I do what you suggest:If I do a search on {gho} I see the list of results, but when I open one entry, I cannot go back. I must re-enter my search and scroll down all the way again. Hitting "back" doesn't help. (am I doing it wrong?) 
<b>3. Screen size:</b> a) Most entries do not use the entire screen (like my example); b) It could be shown only by an option in settings menu; c) it could be a hidden menu, only shown when sliding up (like the menu from the left).
<b>4. Programming issues</b> I agree it makes no sense to manually add a link to the next entry. I thought it may simply be a link to entry_id=[n+1]. Most of the time, this should be some kind of alphabetical order. I don't think it would slow down the software, as the link is only activated when being touched, right? (I'm no programmer, but it should just say "open entry that has a number higher than this entry, if noexist n+1 then try n+2 etc.")

**Comment by De7vID** [comment]
Hitting "back" from an entry you got to through search should bring you back to the search results list. If this isn't happening, something's wrong.

**Comment by LievenLitaer** [comment]
You're right! I tried again - What I meant is the gray search-result-preview. I never push the "show result" button. I knew I was doing something wrong.... I still want my buttons anyway  ;-)

**Comment by LievenLitaer** [comment]
_ping_

**Comment by De7vID** [comment]
I think these would take up too much space. This isn't a problem for you, but some of the devices {boQwI'} supports have much smaller screens than yours.

**Comment by LievenLitaer** [comment]
**grin** - As I have mentioned above, this could easily be done as an option activated in the settings menu: `"show browsing buttons: on/off"`

I know this is no bug and has no priority, but I simply will keep begging like a child for this "feature" ;-)

**Comment by valdr** [comment]
qaptaH Quvar; puq rur...

**Comment by De7vID** [comment]
@valdr - SoH 'Iv?

**Comment by valdr** [comment]
qa'pIn, qorghon puqloD, qI'meQ tuq jIH. 
pongwIj lo'lu' loghjaj ponglu'.

**Comment by LievenLitaer** [comment]
qa'pIn vISov. qepHom jeSwI'na' ghaH. ben law' vIghompu'. ne'Derlan Dab 'ach DoyIchlan Hol jatlhlaH.
Oh, and by the way.... this is my annual reminder for this small improvement wish.  *ping*  ;-)

**Comment by De7vID** [comment]
Fixed in #346.
